### PR TITLE
Car Object Detection Using DL

### PR DESCRIPTION
**Pull request by Rithish5513U**
# Pull Request for DL-Simplified 💡

## Issue Title : Car Object Detection Using DL

- **Info about the related issue (Aim of the project)** : The aim of the project is to detect the cars in the given images.
- **Name:** Rithish S
- **GitHub ID:** https://www.gihub.com/Rithish5513U
- **Email ID:** rithish.satish@gmail.com
- **Idenitfy yourself: (Mention in which program you are contributing in. Eg. For a JWOC 2022 participant it's, `JWOC Participant`)** GSSoc'24 Contributor

Closes: #241 

### Describe the add-ons or changes you've made 📃
I have created 3 models and compared their accuracy for the given task of detecting cars. The models I have trained are Yolov8, mobilenet ssd and vgg16.

## Type of change ☑️

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

I have tested each models based on their accuracy and loss functions.

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by abhisheks008 — CHANGES_REQUESTED** [review]
image

Fix this and also add some EDA images inside the Images folder. Only model implementation parameters are visible, need to add some data visualization images too.

@Rithish5513U

**Comment by Rithish5513U** [comment]
@abhisheks008 First of all sorry for wrong description title, it is because I edited this readme from that one and second what EDA are u mentioning here, there are only images, whether to add images to show bounding boxes over car or some other??

**Comment by abhisheks008** [comment]
> @abhisheks008 First of all sorry for wrong description title, it is because I edited this readme from that one and second what EDA are u mentioning here, there are only images, whether to add images to show bounding boxes over car or some other??

If you have nothing to show in the EDA you can show some dataset images of your choice.

**Comment by abhisheks008** [comment]
Done with the changes?
@Rithish5513U

**Comment by Rithish5513U** [comment]
Yes now check it if any further changes required let me know
@abhisheks008

**Review by abhisheks008 — APPROVED** [review]
Approved @Rithish5513U

**Comment by binguliki** [comment]
@Rithish5513U Hey bro 👋 
I have seen your work , i have small doubt like did you use VGG19 to directly predict the bounding boxes?

**Comment by Rithish5513U** [comment]
> @Rithish5513U Hey bro 👋 I have seen your work , i have small doubt like did you use VGG19 to directly predict the bounding boxes?

No there will be a seperate csv file for bounding boxes. You should draw those lines using cv2 first and then train your model

**Comment by binguliki** [comment]
@Rithish5513U Yeah means your target labels were those annotations given the image right ?

**Comment by binguliki** [comment]
@Rithish5513U Yeah sure , I wanted to know because i had done a similar project of Bone fracture detection model and i found it difficult like the tensorflow object detection API is still under development stage and there arent many pre trained models available. I used YoloV8 of ultralytics and detectron model

**Comment by Rithish5513U** [comment]
> @Rithish5513U Yeah sure , I wanted to know because i had done a similar project of Bone fracture detection model and i found it difficult like the tensorflow object detection API is still under development stage and there arent many pre trained models available. I used YoloV8 of ultralytics and detectron model

Mm I don't know much about detectron model wanna learn about that too if possible
